jwst/associations/asn_gather.py

```diff
@@ -0,0 +1,160 @@
+        The association.
+    """
+    from .load_as_asn import LoadAsAssociation
+    from . import Association
+
+    source_asn_path = Path(association)
+    source_folder = source_asn_path.parent
+    if destination is None:
+        dest_folder = Path('./')
+    else:
+        dest_folder = Path(destination)
+
+    # Create the associations
+    source_asn = LoadAsAssociation.load(source_asn_path)
+    dest_asn = LoadAsAssociation.load(source_asn_path)
+
+    # Create the new association
+    dest_asn['products'] = []
+    for src_product in source_asn['products']:
+        members = [
+            {'expname': src_member['expname'],
+             'exptype': src_member['exptype']}
+            for src_member in src_product['members']
+            if exp_types is None or src_member['exptype'] in exp_types
+        ]
+        if members:
+            product = {
+                'name': src_product['name'],
+                'members': members
+            }
+            dest_asn['products'].append(product)
+
+    if not dest_asn['products']:
+        raise RuntimeError('No products could be gathered.')
+
+    # Copy the members.
+    shellcmd_args = shellcmd.split(' ')
+    src_folder = source_asn_path.parent
+    for product in dest_asn['products']:
+        for member in product['members']:
+            src_path = Path(member['expname'])
+            logger.info(f'*** Copying member {src_path.name}')
+            if str(src_path.parent).startswith('.'):
+                src_path = src_folder / src_path
+            dest_path = dest_folder / src_path.name
+            process_args = shellcmd_args + [str(src_path), str(dest_path)]
+            logger.debug(f'Shell command in use: {process_args}')
+            result = subprocess.run(process_args, capture_output=True, check=True)
+            logger.debug(result.stdout.decode())
+            logger.debug(result.stderr.decode())
+            logger.info('...done')
+            member['expname'] = dest_path.name
+
+    # Save new association.
+    dest_path = dest_folder / source_asn_path.name
+    _, serialized = dest_asn.dump()
+    logger.info(f'Copying the association file itself {dest_path}')
+    with open(dest_path, 'w') as fh:
+        fh.write(serialized)
+
+    # That's all folks
+    return dest_path
+
+
+def from_cmdline(args=None):
+    """Collect asn_gather arguments from the commandline
+
+    Parameters
+    ----------
+    args : [str[,...]]
+        List of arguments to parse
+
+    Returns : dict
+        Dict of the arguments and their values.
+    """
+    import argparse
+
+    parser = argparse.ArgumentParser(
+        description='Gather an association to a new location'
+    )
+
+    parser.add_argument(
+        'association',
+        help='Association to gather.'
+    )
+    parser.add_argument(
+        'destination',
+        help='Folder to copy the association to.'
+    )
+    parser.add_argument(
+        '-t', '--exp-types', default=None, dest='exp_types',
```

Comment:
You know, we never should've called this "exptype" 5 years ago when we invented it, because there's too much potential for confusion with EXP_TYPE. Should've called it "memtype" (member type) instead. But that's water under the bridge ...

Reply:
Have had this thought myself. Though we will probably need to implement backward compatibility, an issue should be filed to make such a change. Will file said issue.